### PR TITLE
[TG-282][fix] Bug with showing hidden devices on the dashboard

### PR DESCRIPTION
**Pull request by Appius**
- Fixed the bug and corresponding test;
- Fixed small issue with Observable<>.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/garage-it/SmartHouse-frontend/pull/137" target="_blank">ReviewNinja

File: src/app/dashboard/dashboard.component.test.ts [hunk 1]
```diff
@@ -25,19 +25,21 @@ describe('Dashboard', () => {
...
-    let mockDevices;
+    let mockVisibleDevices;
 
     beforeEach(() => {
-        sensor = {device: {executor: false, servo: false}};
-        executor = {device: {executor: true, servo: false}, hidden: false};
-        servo = {device: {executor: false, servo: true}};
-        mockDevices = {
+        let sensor = {device: {executor: false, servo: false}, hidden: true};
+        let executor = {device: {executor: true, servo: false}};
```
Comment (ashkanhosseini) [inline src/app/dashboard/dashboard.component.test.ts]: I'd use const instead of let.

Comment (Appius) [reply src/app/dashboard/dashboard.component.test.ts]: !fixed

Comment (ashkanhosseini) [inline src/app/dashboard/dashboard.component.test.ts]: Thanks

File: src/app/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export class DashboardComponent implements OnInit {
 
     ngOnInit() {
         const widgetsSource = this.route.data
-            .flatMap(({widgets}) => Observable.from(widgets.devices));
+            .flatMap(({widgets}) => Observable.from(widgets.devices.filter(device => device.hidden !== true)));
```
Comment (yakimko) [inline src/app/dashboard/dashboard.component.ts:29]:
```
filter(device => !device.hidden)))
```

**Pull request merged**